src/main/kotlin/frc/robot/autonomous/AutoCommands.kt

```diff
@@ -0,0 +1,95 @@
+package frc.robot.autonomous
+
+import com.pathplanner.lib.auto.AutoBuilder
+import edu.wpi.first.math.controller.PIDController
+import edu.wpi.first.math.geometry.Pose2d
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.math.kinematics.ChassisSpeeds
+import edu.wpi.first.wpilibj.DriverStation
+import edu.wpi.first.wpilibj.DriverStation.Alliance
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.Commands
+import frc.robot.CURRENT_MODE
+import frc.robot.Mode
+import frc.robot.lib.distanceFromPoint
+import frc.robot.lib.getPose2d
+import frc.robot.subsystems.drive.Drive
+import frc.robot.subsystems.drive.DriveCommands
+import frc.robot.subsystems.drive.TunerConstants.PATH_CONSTRAINTS
+import frc.robot.swerveDrive
+import frc.robot.vision
+import java.util.function.DoubleSupplier
+import java.util.function.Supplier
+import org.littletonrobotics.junction.Logger
+
+fun alignWithBestVisionTarget(
+    cameraIndex: Int,
+    ySupplier: DoubleSupplier,
+    xSupplier: DoubleSupplier
+): Command =
+    DriveCommands.joystickDriveAtAngle(swerveDrive, ySupplier, xSupplier) {
+        vision.getYawToTarget(cameraIndex)
+    }
+
+fun pathFindToPose(pose: Pose2d): Command =
+    AutoBuilder.pathfindToPoseFlipped(pose, PATH_CONSTRAINTS, 0.0)
+
+fun alignToPose(
+    drive: Drive,
+    robotPose: Supplier<Pose2d>,
+    targetPoseSupplier: Supplier<Pose2d>
+): Command {
+    if (
+        robotPose
+            .get()
+            .distanceFromPoint(targetPoseSupplier.get().translation) >
+            MAX_ALIGNMENT_DISTANCE
+    )
+        return Commands.none()
+
+    val xController = PIDController(3.0, 0.0, 0.2)
+    val yController = PIDController(3.0, 0.0, 0.2)
+    val rotationController =
+        PIDController(
+            1.0,
+            0.0,
+            0.0,
+        )
+    rotationController.enableContinuousInput(-Math.PI, Math.PI)
+
+    return Commands.run({
+        Logger.recordOutput(
+            "PIDsetpoint",
+            getPose2d(
```

Comment:
If you pass all arguments you can use the regular constructor
```suggestion
            Pose2d(
```